packages/jest-circus/src/deadlineTimeout.ts

```diff
@@ -0,0 +1,64 @@
+import {getState} from './state';
+
+export function deadline(): number {
+  const deadline = getState()?.currentlyRunningChildDeadline;
+  if (null === deadline) {
+    throw new Error('bug! no deadline available');
+  }
+  return deadline;
+}
+
+export async function withinDeadline<T>(promise: Promise<T>): Promise<T> {
+  return timeout(promise, deadline() - Date.now());
+}
+
+function isUs(line: string): boolean {
```

Comment:
do we need this at all? `jest-message-util` should already handle filtering out unwanted frames. Might be needed, I haven't tested

Reply:
It's needed for the tests to pass; you end up with `at ../../home/faux/clone/jest/packages/jest-circus/build/deadlineTimeout.js:53:22` in the stack otherwise.

I have no idea whether that could/should be fixed in `jest-message-util`.

Reply:
Would have thought this line caught it: https://github.com/facebook/jest/blob/d393800b0358c79998c003b978ef162a71fe01e9/packages/jest-message-util/src/index.ts#L50

Not worth it to dig into I think

Reply:
https://github.com/facebook/jest/blob/ca47512c77e6a325bca734a35a4ad15a45dee8aa/packages/jest-message-util/src/index.ts#L220-L222

It always keeps the first line, and the first line is going to be inside jest, because we're explicitly throwing.